NullReferenceException when OmniSharp server starts

_From @sidecus on May 16, 2018 4:42_

## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (2.0.0)
Runtime Environment:
 OS Name:     Windows
 OS Version:  6.3.9600
 OS Platform: Windows
 RID:         win81-x64
```

VS Code version: 1.23.1
C# Extension version: 1.15.2

## Steps to reproduce
OmniSharp started and then throws exception:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at OmniSharp.MSBuild.Discovery.Providers.DevConsoleInstanceProvider.GetInstances()
   at OmniSharp.MSBuild.Discovery.MSBuildLocator.GetInstances()
   at OmniSharp.CompositionHostBuilder.RegisterMSBuildInstance(IMSBuildLocator msbuildLocator, ILogger logger)
   at OmniSharp.CompositionHostBuilder.Build()
   at OmniSharp.Stdio.Host..ctor(TextReader input, ISharedTextWriter writer, IOmniSharpEnvironment environment, IConfiguration configuration, IServiceProvider serviceProvider, CompositionHostBuilder compositionHostBuilder, ILoggerFactory loggerFactory, CancellationTokenSource cancellationTokenSource)
   at OmniSharp.Stdio.Driver.Program.<>c__DisplayClass0_1.<Main>b__1()
   at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at OmniSharp.HostHelpers.Start(Func`1 action)
[ERROR] Error: OmniSharp server load timed out. Use the 'omnisharp.projectLoadTimeout' setting to override the default delay (one minute).
```


_Copied from original issue: OmniSharp/omnisharp-vscode#2310_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@ All changes to the project will be documented in this file.
 ## [1.31.0] - _Not Yet Released_
 * Update to Roslyn 2.8.0 packages, adding support for C# 7.3. (PR: [#1182](https://github.com/OmniSharp/omnisharp-roslyn/pull/1182))
 * MSBuild project system no longer stops when a project fails to load. (PR: [#1181](https://github.com/OmniSharp/omnisharp-roslyn/pull/1181)) 
+* Fixed null-reference exception that could be thrown during MSBuild discovery. ([#1188](https://github.com/OmniSharp/omnisharp-roslyn/issues/1188), PR: [#1189](https://github.com/OmniSharp/omnisharp-roslyn/issues/1188))
 
 ## [1.30.1] - 2018-05-11
 * Fixed a 1.30.0 regression that prevented the script project system from working on Unix-based systems (PR: [#1185](https://github.com/OmniSharp/omnisharp-roslyn/pull/1185))

--- a/src/OmniSharp.Host/MSBuild/Discovery/Providers/DevConsoleInstanceProvider.cs
+++ b/src/OmniSharp.Host/MSBuild/Discovery/Providers/DevConsoleInstanceProvider.cs
@@ -36,7 +36,7 @@ namespace OmniSharp.MSBuild.Discovery.Providers
             var versionString = Environment.GetEnvironmentVariable("VSCMD_VER");
             Version.TryParse(versionString, out var version);
 
-            if (version == null)
+            if (version == null && versionString != null)
             {
                 var dashIndex = versionString.IndexOf('-');
 

